Need to force coder when applying transform to map

Had issues when transforming on a map. Fix provided by @jto but should perhaps be integrated/fixed into scio.
```
val data: PCollection[Struct] = sc.pipeline.apply(
  SpannerIO.read()
    .withProjectId(spannerInstanceProject)
    .withInstanceId(spannerInstanceId)
    .withDatabaseId(spannerDatabaseId)
    .withQuery("SELECT * FROM " + spannerTableName))

    val transformedData: SCollection[Transformed] = sc.wrap(data.apply(ParDo.of(new TransformRow())))
```
Reading several rows, as below, from the map breaks
```
val data =
      ids.map(id => ReadOperation.create()
        .withQuery("SELECT * FROM " + spannerTableName + " WHERE id = " + "\"" + id + "\""))
        .applyTransform(SpannerIO.readAll()
        .withProjectId(spannerInstanceProject)
        .withInstanceId(spannerInstanceId)
        .withDatabaseId(spannerDatabaseId))
        .applyTransform(ParDo.of(new TransformRow()))
```
The fix provided by @jto was to fore the Coder after mapping
```
.setCoder(org.apache.beam.sdk.coders.SerializableCoder.of(classOf[ReadOperation])))
```

The problem comes from Kryo not serializing `ReadOperation` properly with causes a NPE:

```
Caused by: com.esotericsoftware.kryo.KryoException: java.lang.NullPointerException
Serialization trace:
keys (com.google.cloud.spanner.KeySet)
keySet (org.apache.beam.sdk.io.gcp.spanner.AutoValue_ReadOperation)
    at com.esotericsoftware.kryo.serializers.ObjectField.write(ObjectField.java:101)
    at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:508)
    at com.esotericsoftware.kryo.Kryo.writeObjectOrNull(Kryo.java:629)
    at com.esotericsoftware.kryo.serializers.ObjectField.write(ObjectField.java:86)
    at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:508)
    at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:651)
```

--- a/scio-core/src/main/scala/com/spotify/scio/coders/instances/JavaCoders.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/coders/instances/JavaCoders.scala
@@ -31,6 +31,7 @@ import org.apache.beam.sdk.io.gcp.pubsub.{PubsubMessage, PubsubMessageWithAttrib
 import org.apache.beam.sdk.io.gcp.bigquery.TableRowJsonCoder
 import org.apache.beam.sdk.transforms.windowing.{BoundedWindow, IntervalWindow, PaneInfo}
 import org.apache.beam.sdk.coders.{Coder => _, _}
+import org.apache.beam.sdk.io.gcp.spanner.ReadOperation
 
 private object VoidCoder extends AtomicCoder[Void] {
   override def encode(value: Void, outStream: OutputStream): Unit = ()
@@ -107,4 +108,8 @@ trait JavaCoders {
 
   implicit def coderJEnum[E <: java.lang.Enum[_]: scala.reflect.ClassTag]: Coder[E] =
     Coder.kryo[E]
+
+  // #1447: Kryo throws a NPE
+  implicit def spannerReadOperationCoder: Coder[ReadOperation] =
+    Coder.beam(bcoders.SerializableCoder.of(classOf[ReadOperation]))
 }

--- a/scio-core/src/test/scala/com/spotify/scio/coders/coders.scala
+++ b/scio-core/src/test/scala/com/spotify/scio/coders/coders.scala
@@ -364,4 +364,10 @@ class CodersTest extends FlatSpec with Matchers {
     check(new FixedSpefificDataExample(bytes))
   }
 
+  it should "#1447: Properly serde spanner's ReadOperation" in {
+    import org.apache.beam.sdk.io.gcp.spanner.ReadOperation
+    val ro = ReadOperation.create().withQuery("SELECT 1")
+    check(ro)
+  }
+
 }

